execution_engine/src/core/engine_state/genesis.rs

```diff
@@ -403,6 +410,12 @@ impl Distribution<ExecConfig> for Standard {
             rng.gen_range(1, 1_000_000_000),
         );
 
+        let state_root_hash = if rng.gen() {
+            Some(Blake2bHash::new(&rng.gen::<[u8; Blake2bHash::LENGTH]>()))
+        } else {
+            None
+        };
```

Comment:
Just curious: Would this work, too?
```rust
let state_root_hash =
    rng.gen().map(|()| Blake2bHash::new(&rng.gen::<[u8; Blake2bHash::LENGTH]>()))
```
(The first `gen()` should return `None` or `Some(())`, with 50% chance each?)

Reply:
Will this work?
```rust
rng::<Option<_>>().map(|bytes| Blake2bHash::new(&bytes))
```
?

Reply:
Sorry, I shouldn't have made that comment — I was just curious!
The code is fine as it is!